workflows: Fix Hubble flow capture in smoke tests

pchaigno opened the pull request:
The step "Capture cilium-sysdump" currently fails in smoke tests with:

    gzip: stdin: not in gzip format
    tar: Child returned status 1
    tar: Error is not recoverable: exiting now
    Error: Process completed with exit code 2.

This is happening because we attempt to download the latest `hubble` binary at URL https://github.com/cilium/hubble/releases/download/latest/hubble-linux-amd64.tar.gz and fail to decompress the downloaded file. That's because the downloaded file contains only "Not found".

Indeed, URL https://github.com/cilium/hubble/releases/download/latest/hubble-linux-amd64.tar.gz doesn't exist, the correct URL is https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz. Never mind that other releases are downloaded at https://github.com/cilium/hubble/releases/download/$TAG/hubble-linux-amd64.tar.gz ¯\_(ツ)_/¯

Fixes: https://github.com/cilium/cilium/pull/16968
/cc @christarazi

tklauser reviewed (commented):
Another option would be to use the latest stable Hubble version per `stable.txt` as documented in https://docs.cilium.io/en/v1.10/gettingstarted/hubble_setup/#install-the-hubble-client

nbusseneau commented on .github/workflows/tests-smoke-ipv6.yaml:
```diff
@@ -133,8 +133,7 @@ jobs:
         run: |
-          version="$(curl -sL0 'https://api.github.com/repos/cilium/hubble/releases/latest' | jq -r .tag_name)"
-          curl -sLO "https://github.com/cilium/hubble/releases/download/${version}/hubble-linux-amd64.tar.gz"
+          curl -sLO https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz
```
If we actually want to download the latest stable Hubble CLI, I recommend we go the `stable.txt` route here.

pchaigno replied:
What's the benefit?

nbusseneau replied:
Just saw your message below: right now it is the case that `latest` is always the latest stable, but I was under the impression we intended have stable and patch releases in Hubble CLI and Cilium CLI in the future. Maybe I am wrong though :D

tklauser replied:
AFAIK Hubble CLI only maintains one stable branch now, currently `v0.8` /cc @rolinh to confirm

So I guess either approach is fine, but we might want to check the `sha256sum` of the downloaded tarball in any case.

tklauser replied:
Another thought: one benefit of using `stable.txt` would be that it matches what we suggest to our users in the Cilum docs.

pchaigno replied:
> I was under the impression we intended have stable and patch releases in Hubble CLI and Cilium CLI in the future.

As long as we have a single stable release at any given point, that wouldn't be an issue.

I'm fine with either approach. I'll switch to using `stable.txt` just because that gives us a bit more control over what version is executed.

> we might want to check the sha256sum of the downloaded tarball in any case.

That's IMO a different concern (and we should make sure we do that for all tarball, not just the two touched in this PR).

pchaigno committed:
workflows: Fix Hubble flow capture in smoke tests

The step "Capture cilium-sysdump" currently fails in smoke tests with:

    gzip: stdin: not in gzip format
    tar: Child returned status 1
    tar: Error is not recoverable: exiting now
    Error: Process completed with exit code 2.

This is happening because we attempt to download the latest hubble
binary at URL github.com/cilium/hubble/releases/download/latest and fail
to decompress the downloaded file. That's because the downloaded file
contains only "Not found".

Indeed, URL cilium/hubble/releases/download/latest doesn't exist, the
correct URL is cilium/hubble/releases/latest/download. Nevermind that
other releases are downloaded at
github.com/cilium/hubble/releases/download/$TAG ¯\_(ツ)_/¯

Fixes: 0cbb855a (".github: Capture hubble flows when smoke test fails")
Signed-off-by: Paul Chaignon <paul@cilium.io>

nbusseneau reviewed (approved):
Guess we're not moving to `stable.txt` yet 🚀
